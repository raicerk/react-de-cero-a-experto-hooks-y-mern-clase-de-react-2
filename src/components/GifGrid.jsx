
import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);
    return (
        <>
            <h3>{category}</h3>
            {isLoading && (<h2>Loading...</h2>)}
            {
                images.map((image) => (<GifItem key={image.id} {...image} />))
            }
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}