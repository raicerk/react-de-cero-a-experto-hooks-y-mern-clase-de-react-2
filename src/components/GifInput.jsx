import { useState } from 'react';
import { PropTypes } from 'prop-types';

export const GifInput = ({ onNewValue }) => {
    const InitialValue = ''
    const [input, setInput] = useState(InitialValue);

    const handlerChange = ({ target: { value } }) => {
        setInput(value)
    }

    const handlerSubmit = (event) => {
        event.preventDefault();
        if (input.trim().length <= 1) return;
        onNewValue(input.trim());
        setInput(InitialValue);
    }

    return (
        <form onSubmit={handlerSubmit} aria-label="form">
            <input
                type="text"
                value={input}
                onChange={handlerChange}
                placeholder="Busca tu gif acÃ¡ y presiona enter"
            />
        </form>
    )
}

GifInput.propTypes = {
    onNewValue: PropTypes.func.isRequired,
}