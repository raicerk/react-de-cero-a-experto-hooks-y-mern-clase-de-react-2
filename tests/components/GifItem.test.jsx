import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/index'

describe('Test de componente <GifItem/>', () => {

    const [title, url] = ['Este es un titulo', 'httpsu://raicerk.cl/'];
    
    test('Se debe dibujar el componente y ser valido con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    })

    test('Se debe comprobar que las propiedades de titulo y url se estÃ©n dibujando nuevamente', () => {
        render(<GifItem title={title} url={url}/>);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Se debe mostrar el titulo', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})