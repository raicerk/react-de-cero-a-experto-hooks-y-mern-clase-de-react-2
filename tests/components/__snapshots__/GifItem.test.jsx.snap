// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de componente <GifItem/> Se debe dibujar el componente y ser valido con el snapshot 1`] = `
<div>
  <div>
    <img
      alt="Este es un titulo"
      src="httpsu://raicerk.cl/"
    />
    <p>
      Este es un titulo
    </p>
  </div>
</div>
`;
